### PR TITLE
feat: Add function to validate an identifier

### LINKED ISSUE #350: [FEATURE] Function to check for validity of a prerelease identifier
# What / Why
Problem described in #349: An 'invalid' identifier will lead to an invalid `semver`.
Now, there is no way to check beforehand whether an identifier is valid, except doing this:
```
const isValidIdentifier(identifier) => !!valid(inc(parse('0.0.1'), 'prerelease', identifier));
```
This is a request for a more straight-forward function, e.g. `identifier()`, which returns `null` (or `false`) if the identifier is invalid.

## References
  * Related to #349

### PR DESCRIPTION
**Pull request by wraithgar**
Cleanup and linting of #351 
Closes #350 

Authored by @boxcee

### CODE REVIEW

File: functions/identifier.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const { t, src } = require('../internal/re')
+const identifier = (version) => {
+  const split = version.split('.')
+  const valid = split.every(part => new RegExp(`^${src[t.PRERELEASEIDENTIFIER]}$`).test(part))
```
Comment (H4ad) [inline functions/identifier.js:4]: For performance reasons, is better to cache this RegExp instead of creating an instance every time.

Comment (wraithgar) [reply functions/identifier.js:4]: I agree. This should exist in `internal/re.js`.

**Comment by wraithgar** [comment]
the codebase calls this a `prereleaseidentifier` which is quite a mouthful

**Comment by ljharb** [comment]
The bin (which is user-exposed, unlike the codebase) calls it `--preid` - perhaps `preID` as short for `preleaseIdentifier`?

**Comment by wraithgar** [comment]
Looking a little deeper at this, the impetus was https://github.com/npm/node-semver/issues/349 which this PR doesn't address.  The line we've been using on these PRs is that if they were ready to go we could land them, but if they needed more work we'd close them and hope that folks who wanted the feature would revisit the work.  I'm gonna close this in light of #349 which is the real bug to be fixed here.

**Comment by ljharb** [comment]
@wraithgar i have a local branch ready to fix #349 once this lands :-/

**Comment by wraithgar** [comment]
Can you implement it without having to export a new function?

**Comment by ljharb** [comment]
Certainly I can, but then consumers won't have an easily accessible predicate function to help explain why they got `null`. What's the argument against adding this?

**Comment by boxcee** [comment]
As I understand the closing comment, this PR needs more work. I cannot see any requests to change anything though, except perhaps the naming.

**Comment by wraithgar** [comment]
In rebasing old PRs the line was drawn at "is this ready" and since it didn't address the underlying issue and folks wanted to bikeshed the name it was deemed "not ready" and closed.

If there were someone actively making a new PR and responding to feedback it would likely be landed.

**Pull request closed without merging**